psydac/linalg/tests/test_stencil_matrix.py

```diff
@@ -3154,6 +3154,50 @@ def test_stencil_matrix_2d_parallel_backend_dot(dtype, n1, n2, p1, p2, sh1, sh2,
     assert (M + M).backend is M.backend
     assert (2 * M).backend is M.backend
 
+@pytest.mark.parametrize('dtype', [float])#, complex])
+@pytest.mark.parametrize('n1', [4,6,10])
+@pytest.mark.parametrize('p1', [3,4])
+@pytest.mark.parametrize('s1', [1,2,3])
+def test_stencil_matrix_1d_toarray(dtype, n1, p1, s1, P1=True):
```

Comment:
There is already a function called `test_stencil_matrix_1d_serial_toarray`. Why not parametrizing that one instead of writing a new function?

Reply:
See [here](https://github.com/pyccel/psydac/blame/7e621bb5dc82b60c8e460680d5262965424e5a45/psydac/linalg/tests/test_stencil_matrix.py#L433-L438):
```python
@pytest.mark.parametrize('dtype', [float])
@pytest.mark.parametrize('n1', [7])
@pytest.mark.parametrize('p1', [1,2])
@pytest.mark.parametrize('s1', [1])
@pytest.mark.parametrize('P1', [True, False])
def test_stencil_matrix_1d_serial_toarray( dtype, n1, p1, s1, P1):
    ...
```

Reply:
You're right, I didn't see it at first. There is also a test for 2d in `test_stencil_matrix_2d_serial_toarray`, I will parameterize both.

Should I leave the 3d kernels as they are until we have a `test_stencil_matrix_2d_serial_toarray` unit test too?